### ISSUE TITLE
Is Swashbuckle.AspNetCore.ReDoc being maintaned?

### ISSUE THREAD

**Issue by LTMenezes** [domaindrivendev/Swashbuckle.AspNetCore#738]
Hi!

I currently use Swagger and Redoc on .Net Framework, I looked at the source code of this project and realized there's already a built-in Redoc package. However I'm getting a error trying to use it, the Redoc webpage goes to infinite loading and at the browser console you can see the error: "redoc.min.js:7 ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'toString' of undefined
TypeError: Cannot read property 'toString' of undefined".

I have tried looking at the documentation for details but I haven't found any regarding the Redoc package inside swashbuckle, is the development of this project currently active?

**Comment by RehanSaeed** [comment]
Did you find an answer?

**Comment by JonPSmith** [comment]
I'm also trying to use ReDoc. I installed the package `Swashbuckle.AspNetCore.ReDoc` and replaced UseSwagger with UseRedoc, but I couldn't find a page to go to, i.e. localhost:xxxxx/swagger showed page not found.

Is there any documentation on how to use ReDoc?

**Comment by cnkeenan** [comment]
@JonPSmith Did you point the SpecUrl after setting ReDoc as the UI to the swagger.json location?

**Comment by JonPSmith** [comment]
Yes, I set SpecUrl to the same setting that SwaggerUi used, and I set the DocumentTitle to the same as SwaggerUi - see code below
```csharp
            app.UseReDoc(c =>
            {
                c.SpecUrl = "/swagger/v1/swagger.json";
                c.DocumentTitle = apiTitle;
            });
```
I got a 404 status when I went to https://localhost:xxxxx/swagger

**Comment by domaindrivendev** [comment]
I believe the default RoutePrefix for ReDoc is “/api-docs” as opposed to “swagger”.

You’re correct that the docs are lacking - are you interested in submitting a PR to update them?

**Comment by JonPSmith** [comment]
OK,

Yes https://localhost:xxxxx/api-docs works, thanks. Unfortunately once I saw ReDoc it only provides the docs, but no ability (I could see) like SwaggerUI's "Try it out" where you can exercise the ASP.NET Core's Web API controllers.

To help others who want to try ReDoc then you need to:

1. Load the NuGet package Swashbuckle.AspNetCore.ReDoc
2. replace the `app.UseSwaggerUI(...` in your ASP.NET Core Startup class with
```c#
app.UseReDoc(c =>
{
     c.SpecUrl = "/swagger/v1/swagger.json";
     c.DocumentTitle = "Your API name";
 });
```
3. Go to https://localhost:xxxxx/api-docs to see the ReDoc pages

NOTE to @domaindrivendev. I was really looking for a more compact UI, like https://github.com/jensoleg/swagger-ui, which includes "Try it out" feature. I saw your issue #377, but that looked like hard work.

**Comment by lonix1** [comment]
@JonPSmith (or anyone else)

Did you ever manage to integrate swashbuckle with another swagger-ui theme/fork, like [https://github.com/jensoleg/swagger-ui](https://github.com/jensoleg/swagger-ui)?

**Comment by JonPSmith** [comment]
Hi @lonix1,

No, I gave up. Sorry.

**Comment by cnkeenan** [comment]
@lonix1 are you asking if anyone has implemented the Try It Out feature seen in SwaggerUI in the swashbuckle version of ReDocUI?

**Comment by lonix1** [comment]
No I'm asking if anyone has managed to use a different ui, like discusses above. I assume not. Also I see that those other uis have not been maintained in a while, so it might be pointless to try. swaggerui is ugly but at least it works.

**Comment by cnkeenan** [comment]
Well the ReDoc UI is a different UI than swagger ui, but the functionality
of swagger try it out feature does not exist in ReDoc yet

**Comment by JonPSmith** [comment]
@lonix1 

> swaggerui is ugly but at least it works.

That was the conclusion I came to as well.

**Comment by sommmen** [comment]
app.UseSwaggerUi3(); // serve Swagger UI
            app.UseReDoc(); // serve ReDoc UI

For other uses, by default - both use /swagger.

**Comment by GitItKrishna** [comment]
I did the below steps : For ReDoc, i added a RoutePrefix, this way i was able to use both UseSwaggerUi and UseReDoc 

1. Load the NuGet package Swashbuckle.AspNetCore.ReDoc
2.  app.UseSwaggerUI(
                
                });
3. app.UseReDoc(c =>
            {
                c.RoutePrefix = "api/redoc";
                c.DocumentTitle = "API REDOC Documentation with OpenAPI, Swashbuckle";
                c.SpecUrl = "/api/v1/swagger.json";
            });

For Using SwaggerUI, i browsed the url 'http://localhost:5286/api/index.html'
For using Redoc, I browsed the url 'http://localhost:5286/api/redoc/index.html'
